### PR TITLE
python3Packages.dbt-core: init at 1.0.0

### PR DESCRIPTION
**Pull request by shikanime**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dbt/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  meta = with lib; {
+    description = ''
+      dbt (data build tool) is a command line tool that helps analysts and
+      engineers transform data in their warehouse more effectively
+    '';
+    homepage = "https://pypi.python.org/pypi/dbt/";
+    license = licenses.mit;
+    platforms = platforms.all;
```
Comment (fabaff) [inline pkgs/development/python-modules/dbt/default.nix:21]: Can be removed.

```suggestion
```

File: pkgs/development/python-modules/dbt/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "276ced7e9e3cb22e5d7c14748384a5cf5d9002257c0ed50c0e075b68011bb6d0";
+  };
+
+  meta = with lib; {
+    description = ''
+      dbt (data build tool) is a command line tool that helps analysts and
+      engineers transform data in their warehouse more effectively
```
Comment (fabaff) [inline pkgs/development/python-modules/dbt/default.nix:17]: It's a module and CLI tool.

```suggestion
    description = "Python module to transform data";
```

"Extensive" content should go to `longDescription`.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -13172,4 +13172,10 @@
     github = "sei40kr";
     githubId = 11665236;
   };
+  shikanime = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:13175]: Adding a new entry to `maintainer-list.nix ` should be a separate commit in this PR:

File: pkgs/development/python-modules/dbt/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{ lib, fetchPypi, buildPythonPackage, autopep8 }:
+
+buildPythonPackage rec {
+  pname = "dbt";
+  version = "0.19.0";
+
```
Comment (fabaff) [inline pkgs/development/python-modules/dbt/default.nix:6]: Only [Python 3.7 and above](https://github.com/dbt-labs/dbt-core/blob/main/setup.py#L83) is supported.

```suggestion
  version = "0.19.0";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/dbt/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  propagatedBuildInputs = [ autopep8 ];
```
Comment (fabaff) [inline pkgs/development/python-modules/dbt/default.nix:7]: This package should [depend](https://github.com/dbt-labs/dbt-core/blob/main/setup.py#L24-L26) on `dbt-core`, `dbt-postgres`, `dbt-redshift`, `dbt-snowflake` and `dbt-bigquery`. `dbt` is a meta-package nowadays.

The `setup.py` file also reflect that:

```text
    install_requires=[
        'dbt-core=={}'.format(package_version),
        'dbt-postgres=={}'.format(package_version),
        'dbt-redshift=={}'.format(package_version),
        'dbt-snowflake=={}'.format(package_version),
        'dbt-bigquery=={}'.format(package_version),
    ],
```

File: pkgs/development/python-modules/dbt/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    inherit pname version;
+    sha256 = "276ced7e9e3cb22e5d7c14748384a5cf5d9002257c0ed50c0e075b68011bb6d0";
+  };
+
+  meta = with lib; {
+    description = ''
+      dbt (data build tool) is a command line tool that helps analysts and
+      engineers transform data in their warehouse more effectively
+    '';
+    homepage = "https://pypi.python.org/pypi/dbt/";
```
Comment (fabaff) [inline pkgs/development/python-modules/dbt/default.nix:19]: I would suggest to actually point to upstream's repo (https://github.com/dbt-labs/dbt-core) or the documentation (https://docs.getdbt.com/docs/).

File: pkgs/development/python-modules/dbt-bigquery/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "dbt-bigquery";
+  version = "0.21.0";
+
+  disabled = pythonOlder "3.7";
+
+  propagatedBuildInputs = [
+    autopep8
```
Comment (risicle) [inline pkgs/development/python-modules/dbt-bigquery/default.nix:20]: Is this actually a _runtime_ requirement for the package? Not just needed for e.g. the tests?

**Comment by risicle** [comment]
Fails for me on nixos x86_64:

```
builder for '/nix/store/vwqy469lywhw8vkz9s8sgb80lvg383ha-python3.9-dbt-core-0.21.0.drv' failed with exit code 1; last 10 log lines:
  Processing ./autopep8-1.5.7-py2.py3-none-any.whl
  Requirement already satisfied: toml in /nix/store/2ixdjrpazvyr1zr1x8yi566j1rrn4vl0-python3.9-toml-0.10.2/lib/python3.9/site-packages (from autopep8==1.5.7) (0.10.2)
  Requirement already satisfied: pycodestyle>=2.7.0 in /nix/store/b7l7nb3i8gk53dihhs4hixg7hxwxgdsa-python3.9-pycodestyle-2.8.0/lib/python3.9/site-packages (from autopep8==1.5.7) (2.8.0)
  Installing collected packages: autopep8
    Attempting uninstall: autopep8
      Found existing installation: autopep8 1.6.0
      Uninstalling autopep8-1.6.0:
  ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: '/nix/store/ardhhrlqwrcf7c8kgdpmcwy1pslbzbdj-python3.9-autopep8-1.6.0/bin/autopep8'
  Consider using the `--user` option or check the permissions.

```

File: pkgs/development/python-modules/dbt-bigquery/default.nix (hunk 7 above)
Comment (Shikanime) [reply pkgs/development/python-modules/dbt-bigquery/default.nix:20]: Yes in the setup.py source code (`python_requires=">=3.7"`)

**Comment by Shikanime** [comment]
Mhhh, I'm not completely sure what the problem is but I have an idea, I'll take a look at it tonight.

File: pkgs/development/python-modules/dbt-bigquery/default.nix (hunk 7 above)
Comment (risicle) [reply pkgs/development/python-modules/dbt-bigquery/default.nix:20]: No, I'm referring to `autopep8`

Comment (Shikanime) [reply pkgs/development/python-modules/dbt-bigquery/default.nix:20]: TBH, I'm not sure, it seems that only the test phase uses it, but I can't find anything about autopep8 or pycodestyle in DBT and all dependencies.

**Comment by Shikanime** [comment]
@risicle By any chance, do you have any idea about my current issue: the construction passes but the store always seems to be empty, I'm not sure how to interpret it.

**Comment by Shikanime** [comment]
@fabaff This is my first contribution, I'm now facing a dependency constraint issue with the package I'm integrating into nixpkgs, is there anything I can do or any contributor documentation I should look at?

```log
Processing ./dbt_core-0.21.0-py3-none-any.whl
ERROR: Could not find a version that satisfies the requirement json-rpc<2,>=1.12 (from dbt-core) (from versions: none)
ERROR: No matching distribution found for json-rpc<2,>=1.12
error: builder for '/nix/store/ymn69fh0g1va7rfm34j7yv301dk9pa5a-python3.9-dbt-core-0.21.0.drv' failed with exit code 1;
       last 10 log lines:
       > adding 'dbt_core-0.21.0.dist-info/top_level.txt'
       > adding 'dbt_core-0.21.0.dist-info/RECORD'
       > removing build/bdist.macosx-10.6-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /private/tmp/nix-build-python3.9-dbt-core-0.21.0.drv-1/dbt-core-0.21.0/dist /private/tmp/nix-build-python3.9-dbt-core-0.21.0.drv-1/dbt-core-0.21.0
       > Processing ./dbt_core-0.21.0-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement json-rpc<2,>=1.12 (from dbt-core) (from versions: none)
       > ERROR: No matching distribution found for json-rpc<2,>=1.12
       For full logs, run 'nix log /nix/store/ymn69fh0g1va7rfm34j7yv301dk9pa5a-python3.9-dbt-core-0.21.0.drv'.
```
